api/src/main/resources/workbench-api.yaml

```diff
@@ -0,0 +1,6369 @@
+---
+swagger: '2.0'
+info:
+  version: 0.1.0
+  title: AllOfUs Workbench API
+  description: The API for the AllOfUs workbench.
+  termsOfService: http://www.pmi-ops.org/terms_of_service.html
+  contact:
+    name: developer_help@pmi-ops.org
+  license:
+    name: BSD
+host: api.pmi-ops.org
+securityDefinitions:
+  aou_oauth:
+    authorizationUrl: ''
+    tokenUrl: ''
+    type: oauth2
+    flow: accessCode
+schemes:
+- https
+produces:
+- application/json
+security:
+- aou_oauth: []
+parameters:
+  userId:
+    in: path
+    name: userId
+    type: integer
+    format: int64
+    required: true
+  conceptSetId:
+    in: path
+    name: conceptSetId
+    type: integer
+    format: int64
+    required: true
+    description: Concept set ID
+  dataSetId:
+    in: path
+    name: dataSetId
+    type: integer
+    format: int64
+    required: true
+    description: Data set ID
+  workspaceNamespace:
```

Comment:
My current thinking (not implemented yet) is to find or define a canonical Swagger ordering and always save the file to that. This would make it easier to keep the format consistent and avoid conflicts, by standardizing the quote style and always ordering the keys alphabetically. Probably drop all comments and encourage folks to use description fields or other mechanisms built into the structure instead.

On the other hand, this makes the file feel less like source code and more like a serialization format. In reality it's both, but treating it like the latter makes it easier to process for us, and keeps folks from expecting untethered text to stick to following nodes.

Reply:
I'm not sure I follow, are you trying to build a custom autoformatter for this file? Why would we not treat this as a source file going forward? Why the concern about having things "untethered" to nodes? I assumed this was a one-time issue while you are attempting to automatically merge files and migrate us to OpenAPI 3.

Comments have a role in the source document that is separate from the API being documented, e.g. to establish conventions, track TODOs, explain Swagger-related hacks, noting to keep things in sync.

Reply:
I buy that we could move some of these comments into a special yaml properties, but I don't think my above question has been answered yet - i.e. why is this actually a problem?

Also note the scale here is 3-4 comments, so I don't think we should spend too much time automating something here.

Reply:
Sorry, missed these.

It's a subtle difference, but by letting go of comments, we can tighten up semantics.

Basically, I wanted license to use automatic, yaml- and swagger-processing tools to operate on this file for a few purposes, then save it into a canonical ordering (so manual edits continue working). The requirement to keep comments around makes both of those impossible, as they're not supported in Swagger per se, and only supported insofar as they're dropped on the floor in YAML. 

Examples of processes/tools requiring round tripping I was thinking about in addition to OpenAPI 3:
-  A structural linter that would lay out sections in the order we want and alphabetize the rest (in cases where ordering is arbitrary). I'd have to see if any string formatters exist that could apply readability rules to string definitions.
- A script/tool to split the file into subdirectories and reusable YAML files according to rules we decide on
- Custom tooling for us to validate that our APIs conform to our own standards in addition to OpenAPIs, that we don't omit critical tags or defaults, etc. (This would only need round-tripping if it had an "autofix" feature).

Comments absolutely have a role, but the issue is that in true code files, because relative order of lines is important in most places, and you only ever go in one direction, you don't have to worry about losing them. File types like XML, JSON, and JSON don't have as much directionality to them, and can be transformed in interesting ways by processors that only care about the semantics of the file. Comments like these live in the text file describing the document, not the document itself. Line numbers, text wrapping, quote styles, and inline comments seemed like a reasonable price to pay to me.